src/test/java/core/basesyntax/RegistrationTests.java

```diff
@@ -0,0 +1,145 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationIllegalArgumentException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationTests {
+    public static final String ANSI_YELLOW = "\u001B[33m";
+    public static final String ANSI_CYAN = "\u001B[36m";
+    public static final String ANSI_RESET = "\u001B[0m";
+    private static final long ID1 = 1L;
+    private static final long ID2 = 2L;
+    private static final int NON_ACCEPTABLE_AGE = 9;
+    private static final int ACCEPTABLE_AGE = 34;
+    private static final String EMPTY_LOGIN = "";
+    private static final String CORRECT_LOGIN = "Ukraine";
+    private static final String INCORRECT_PASSWORD = "34df";
+    private static final String CORRECT_PASSWORD = "GloryOfUkraine";
+    private static RegistrationServiceImpl registrationService;
+    private static StorageDaoImpl storageDao;
+    private static User validUser = new User();
+    private static User invalidUser = new User();
+
+    @BeforeAll
+    static void setUp() {
+        System.out.println(ANSI_YELLOW
+                + "----------------------  Tests started  ----------------------" + ANSI_RESET);
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl();
+        validUser.setId(ID1);
+        validUser.setAge(ACCEPTABLE_AGE);
+        validUser.setLogin(CORRECT_LOGIN);
+        validUser.setPassword(CORRECT_PASSWORD);
+        invalidUser.setId(ID2);
+    }
+
+    @BeforeEach
+    void initialUsers() {
+        invalidUser.setAge(NON_ACCEPTABLE_AGE);
+        invalidUser.setLogin(EMPTY_LOGIN);
+        invalidUser.setPassword(INCORRECT_PASSWORD);
+    }
+
+    @Test
+    @DisplayName("Registration a NULL User object")
+    public void register_null_user_NotOk() {
+        assertThrows(RegistrationIllegalArgumentException.class, () -> {
+            registrationService.register(null); }, "Expected "
+                + RegistrationIllegalArgumentException.class.getName()
+                + " to be thrown for the null user login, but it wasn't");
+        System.out.println("Test register_null_user_NotOk passed");
+    }
+
+    @Test
+    @DisplayName("Registration an user with null login")
+    public void register_user_with_null_login_NotOk() {
+        validUser.setLogin(null);
```

Comment:
You set login = null for validUser and registering invalidUser, please check this method

Reply:
auto mistake. thank's